R/restopt_component.R

```diff
@@ -0,0 +1,30 @@
+#' @include internal.R
+NULL
+
+#' Restoration problem component
+#'
+#' Create a component to add to a restoration optimization problem.
+#'
+#' @param name `character` Name of the component.
+#'
+#' @param java `character` Name Java method for adding component.
+#'
+#' @param class `character` Vector of classes to identify the component.
+#'
+#' @param data `list` Object containing parameters for adding the component.
+#'  Defaults to an empty list.
+#'
+#' @param args `character` Vector used to map the names of the elements in
+#'  `data` to the Java method.
+#'  Defaults to an empty chracter vector.
+#'
+#' @return An object.
+#'
+#' @examples
+#' # TODO
+#'
+#' @export
```

Comment:
So this is the generic function to create a component (constraint, objective, ...), and it contains the name of the corresponding java method to call, and its arguments, am I right?

Reply:
Also, I am wondering, is it really useful to have such a file, as it will never be used, and it does not factorize any code?

Reply:
Yeah that's right. I was thinking that this class/function/constructor could be useful to store data/information for each constraint/objective in a standard way. After sotring the data/information for an arbitrary constraint/objective in a standard way, I was thinking that the `solve()` function could read the information/data stored in these objects and use them to build the problem specification using the Java API. I was also thinking another advantage of this class/function/constructor is that it could be run data sanity checks (e.g. ensure that the `java` parameter contains a valid `character` value) to minimize chances of passing invalid values to the Java API. Does that make sense? I completely agree that we want to avoid bloating the code - maybe there's a better way to do this?

Reply:
Actually, this function is useful and do factorize code, I was mistaken in the first place.

Reply:
Ok great - I think we're in agrement, so I'll mark this comment thread as resolved